lib/modulesync/git.rb

```diff
@@ -100,6 +110,7 @@ def self.update(name, files, options)
       module_root = "#{options[:project_root]}/#{name}"
       message = options[:message]
       repo = ::Git.open(module_root)
+      options[:branch] ||= (repo.current_branch || 'master')
```

Comment:
I dislike changing options this way and think a local variable is cleaner. Same below.

Reply:
You're right, that's crap.  I started to change it to a local variable and then put it back to this for some reason.  I know that I had some questions about why the branch is being checked out again here and in `update_noop` after `switch_branch` is called in `pull`.  Looking over the code in `ModuleSync::update` again I suppose `ModuleSync::Git::update_noop` could be called in offline mode so you may need to check out a branch then.